backend/coreapp/compiler_wrapper.py

```diff
@@ -156,6 +156,71 @@ def compile_code(
 
+            # Hack to get mac compiler output to work with asm_differ
+            if compiler.is_mac_compiler:
+                asm_dump_cmd = compiler.asm_dump
+                convert_syntax = compiler.assemble_cmd
+                try:
+                    compile_proc = sandbox.run_subprocess(
+                        asm_dump_cmd,
+                        mounts=[compiler.path],
+                        shell=True,
+                        env={
+                            "PATH": PATH,
+                            "WINE": WINE,
+                            "INPUT": sandbox.rewrite_path(object_path),
+                            "OUTPUT": sandbox.rewrite_path(sandbox.path / "c_asm.s"),
+                            "COMPILER_DIR": sandbox.rewrite_path(compiler.path),
+                            "FUNCTION": function,
+                            "MWCIncludes": "/tmp",
+                        },
+                    )
+                except subprocess.CalledProcessError as e:
+                    # Compilation failed
+                    if e.stdout:
+                        msg = f"{e.stdout}\n{e.stderr}"
+                    else:
+                        msg = e.stderr
+
+                    logging.debug("Compilation failed: %s", msg)
+                    raise CompilationError(e.stderr)
+                except ValueError as e:
+                    # Shlex issue?
+                    logging.debug("Compilation failed: %s", e)
+                    raise CompilationError(str(e))
+                try:
+                    compile_proc = sandbox.run_subprocess(
+                        convert_syntax,
+                        mounts=[compiler.path],
+                        shell=True,
+                        env={
+                            "PATH": PATH,
+                            "WINE": WINE,
+                            "INPUT": sandbox.rewrite_path(sandbox.path / "c_asm.s"),
+                            "OUTPUT": sandbox.rewrite_path(object_path),
+                            "COMPILER_DIR": sandbox.rewrite_path(compiler.path),
+                            "FUNCTION": function,
+                            "MWCIncludes": "/tmp",
+                        },
+                    )
+                except subprocess.CalledProcessError as e:
+                    # Compilation failed
+                    if e.stdout:
+                        msg = f"{e.stdout}\n{e.stderr}"
+                    else:
+                        msg = e.stderr
+
+                    logging.debug("Compilation failed: %s", msg)
+                    raise CompilationError(e.stderr)
+                except ValueError as e:
+                    # Shlex issue?
+                    logging.debug("Compilation failed: %s", e)
+                    raise CompilationError(str(e))
+
+            with open(sandbox.path / "c_asm.s", "r") as file:
+                for line in file:
+                    print("# %s" % line)
+
```

Comment:
I've mentioned before, but this shouldn't be necessary and causes a lot of tech debt wrt duplicated code that needs to be cleaned up now in two places.

At the very least, the common code should be abstracted. But I don't understand why we need two compiler invocations in the first place